tensorboard/notebook.py

```diff
@@ -0,0 +1,336 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities for using TensorBoard in notebook contexts, like Colab."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import datetime
+import shlex
+import sys
+
+from tensorboard import manager
+
+
+# See `_get_context` for the meanings of these values.
+_CONTEXT_COLAB = "_CONTEXT_COLAB"
+_CONTEXT_IPYTHON = "_CONTEXT_IPYTHON"
+_CONTEXT_NONE = "_CONTEXT_NONE"
+
+
+def _get_context():
+  """Determine the most specific context that we're in.
+
+  Returns:
+    _CONTEXT_COLAB: If in Colab with an IPython notebook context.
+    _CONTEXT_IPYTHON: If not in Colab, but we are in an IPython notebook
+      context (e.g., from running `jupyter notebook` at the command
+      line).
+    _CONTEXT_NONE: Otherwise (e.g., by running a Python script at the
+      command-line or using the `ipython` interactive shell).
+  """
+  # In Colab, the `google.colab` module is available, but the shell
+  # returned by `IPython.get_ipython` does not have a `get_trait`
+  # method.
+  try:
+    import google.colab
+    import IPython
+  except ImportError:
+    pass
+  else:
+    if IPython.get_ipython() is not None:
+      # We'll assume that we're in a notebook context.
```

Comment:
Nit, I understand what you mean, but the comment could be a hair clearer and would be consistent with the return value if you wrote "... we're in a Colab notebook context."

Reply:
Sure.